src/backend/cuda/jit.cpp

```diff
@@ -105,15 +105,53 @@ static string getKernelString(string funcName, Node *node, bool is_linear)
         kerStream << "%tidy = call i32 @llvm.nvvm.read.ptx.sreg.tid.y()\n";
         kerStream << "%bdmy = call i32 @llvm.nvvm.read.ptx.sreg.ntid.y()\n";
 
-        kerStream << "%id2 = sdiv i32 %bidx, %blkx\n";
-        kerStream << "%id3 = sdiv i32 %bidy, %blky\n";
-        kerStream << "%id2m = mul i32 %id2, %blkx\n";
-        kerStream << "%id3m = mul i32 %id3, %blky\n";
-        kerStream << "%blk_x = sub i32 %bidx, %id2m\n";
-        kerStream << "%blk_y = sub i32 %bidy, %id3m\n";
-        kerStream << "%id0m = mul i32 %blk_x, %bdmx\n";
-        kerStream << "%id1m = mul i32 %blk_y, %bdmy\n";
+        kerStream << "%blk_x = alloca i32, align 4\n";
```

Comment:
It would be nice if you broke this down into function or comment each of the blocks